addons/mail/static/src/components/chat_window/chat_window.xml

```diff
@@ -25,7 +25,7 @@
                         <ThreadViewer
                             class="o_ChatWindow_thread"
                             composerAttachmentsDetailsMode="'card'"
-                            hasComposer="chatWindow.thread.model !== 'mail.box'"
+                            hasComposer="chatWindow.thread.model !== 'mail.box' and (!chatWindow.thread.mass_mailing or env.messaging.device.isMobile)"
```

Comment:
`useStore` is currently not observing `chatWindow.thread` which can cause render issues here (btw that would already have been a potential problem before the PR seeing how thread is used in this template)

Reply:
Btw it feels like "hasComposer" could (should?) be a computed field on thread(viewer?) instead?

Probably out of scope, similarly to `isCompact` discussed in other comment.

Reply:
> Btw it feels like "hasComposer" could (should?) be a computed field on thread(viewer?) instead?
> 
> Probably out of scope, similarly to `isCompact` discussed in other comment.

Yeah, component props are in peril, models and fields are the trend...

Nah seriously, it's weird to see the amount of OWL features we tend to deny: `useState` is bad, component lifecycles is bad, asynchronous in component is bad, cross-component communication is bad... and now props are bad?

I'm intentionally provoking, but all these decisions make a lot of sense when data (knowledge and) management is essential.

Reply:
Well... All of those features have their uses, but not when store and models are involved.
I think in our scope we just never have an opportunity to see them when they shine.

Reply:
Did you push change? I don't see any change in useStore of chat window

Reply:
Totally forgot sorry